syntax = "proto3";
package rtcl_p3s7_control;

service RtclP3s7Control {
      rpc WriteSysReg (WriteRegRequest) returns (BoolResponse);
      rpc ReadSysReg (ReadRegRequest) returns (ReadRegResponse);
      rpc WriteTimgenReg (WriteRegRequest) returns (BoolResponse);
      rpc ReadTimgenReg (ReadRegRequest) returns (ReadRegResponse);
      rpc WriteCamReg (WriteRegRequest) returns (BoolResponse);
      rpc ReadCamReg (ReadRegRequest) returns (ReadRegResponse);
      rpc WriteSensorReg (WriteRegRequest) returns (BoolResponse);
      rpc ReadSensorReg (ReadRegRequest) returns (ReadRegResponse);
      rpc RecordImage (RecordImageRequest) returns (BoolResponse);
      rpc ReadImage (ReadImageRequest) returns (ReadImageResponse);
      rpc RecordBlack (RecordImageRequest) returns (BoolResponse);
      rpc ReadBlack (ReadImageRequest) returns (ReadImageResponse);
}


message Empty {
}

message BoolResponse {
    bool result = 1;
}

message WriteRegRequest {
    uint64 addr = 1;
    uint64 data = 2;
}

message ReadRegRequest {
    uint64 addr = 1;
}

message ReadRegResponse {
    bool result = 1;
    uint64 data = 2;
}

message RecordImageRequest {
    uint64 width = 1;
    uint64 height = 2;
    uint64 frames = 3;
}

message ReadImageRequest {
    uint64 addr = 1;
    uint64 size = 2;
}

message ReadImageResponse {
    bool result = 1;
    bytes image = 5;
}

